snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -90,5 +121,78 @@ def launch_instance(self) -> None:
     def setup_snapcraft(self) -> None:
         self.echoer.info('Setting up snapcraft in {!r}'.format(
             self.instance_name))
+        if common.is_snap():
+            self._inject_snapcraft()
+        else:
+            self._install_snapcraft()
+
+    def _install_snapcraft(self) -> None:
+        """Install snapcraft from the store."""
         install_cmd = ['sudo', 'snap', 'install', 'snapcraft', '--classic']
         self._run(install_cmd)
```

Comment:
This looks like it could be using `_inject_snap` instead, which seems a little odd naming-wise.

Reply:
I don't understand this, but maybe because of the naming, I didn't pay much atention to it before grabbing it from the lxd package.